src/t8_cmesh/t8_cmesh_vertex_conn_tree_to_vertex.hxx

```diff
@@ -0,0 +1,107 @@
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_cmesh_vertex_conn_tree_to_vertex.hxx
+ * Class to save data structure for tree_to_vertex_lists of the cmesh.
+ * When the cmesh stores global vertex numbers, we require a lookup that
+ * matches a tree and its local vertex to a global vertex id.
+ * This lookup is encoded in the t8_cmesh_vertex_conn_tree_to_vertex struct.
+ */
+
+/* TODO:
+ *  It is probably best to set all global ids of a single tree as one attribute.
+ * That way we can store it as a single arrays of id's.
+ * We will probably most often want to access all ids of one tree, so a function returning an array of ids
+ * is a good idea anyway and if we already store them as such then we do not need to do any data movement
+ * when accessing.
+ *
+ * On the downside we will only have a "set all ids of a tree" function and no "set this single id for this tree and this vertex" function.
+ */
+
+#ifndef T8_CMESH_VERTEX_CONN_TREE_TO_VERTEX_HXX
+#define T8_CMESH_VERTEX_CONN_TREE_TO_VERTEX_HXX
+
+#include <algorithm>
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_vertex_conn_vertex_to_tree.hxx>
+
+/* forward declaration of ttv class needed since the two class headers include each other. */
+class t8_cmesh_vertex_conn_vertex_to_tree;
+
+class t8_cmesh_vertex_conn_tree_to_vertex {
+ public:
+  /** Standard constructor. Does nothing. */
+  t8_cmesh_vertex_conn_tree_to_vertex (): state (EMPTY)
+  {
+  }
+
+  /** Constructor from a cmesh where all the attributes are set. */
+  t8_cmesh_vertex_conn_tree_to_vertex (const t8_cmesh_t cmesh)
+  {
+    SC_ABORT ("not implemented.");
+  }
+
+  /** Constructor from a cmesh and a given vertex to tree connectivity.
+   *
+   * \note \a cmesh must not be committed.
+   * \note \a vtt must be committed.
+   * \note This does not work until issue #923 https://github.com/DLR-AMR/t8code/issues/923 is resolved.
+   */
+  t8_cmesh_vertex_conn_tree_to_vertex (const t8_cmesh_t cmesh_from, const t8_cmesh_t cmesh,
+                                       const struct t8_cmesh_vertex_conn_vertex_to_tree &vtt);
+
+  /* Setter functions */
+  /** Set all global vertex ids of a local tree.
+   * \param [in] cmesh The considered cmesh
+   * \param [in] local_tree A local tree id of \a cmesh
+   * \param [in] global_vertex_id The ids of the global vertices in order of \a local_tree's vertices.
+   * \param [in] num_vertices Must match the number of vertices of \a local_tree
+   *
+   * \note \a cmesh must not be committed.
+  */
+  void
+  set_global_vertex_ids_of_tree_vertices (const t8_cmesh_t, const t8_gloidx_t global_tree,
+                                          const t8_gloidx_t *global_tree_vertices, const int num_vertices);
```

Comment:
```suggestion
  template <t8_forward_iterator<t8_gloidx_t> TForwardIt>
  void
  set_global_vertex_ids_of_tree_vertices (const t8_cmesh_t, const t8_gloidx_t global_tree,
                                          const TForwardIt global_tree_vertices_begin, const TForwardIt global_tree_vertices_end);
```

with a `t8.hxx` containing
```cpp
  template <typename TValueType>
  concept t8_forward_iterator = std::forward_iterator<TValueType>;
```

Reply:
Okay until this gets looked over again, `src/t8_types/t8_type.hxx` will be merged. Then we can put it there instead of creating a `t8.hxx`